### ISSUE TITLE
C-c C-k after cider-jack-in-clojurescript sends request to both repls

### ISSUE THREAD

**Issue by expez** [clojure-emacs/cider#1264]
Steps to reproduce:

1. cider-jack-in-clojurescript
2. <kbd>C-c C-k</kbd> in a buffer with `:require` clause with prefix notation
3. Check the cljs repl and you should see: 
```
clojure.lang.ExceptionInfo: Only :as and :refer options supported in :require / :require-macros; offending spec: [cider.nrepl.middleware.util [cljs :as cljs] [misc :refer [err-info]]] {:tag :cljs/analysis-error}
	at clojure.core$ex_info.invoke(core.clj:4593)
	at cljs.analyzer$error.invoke(analyzer.cljc:525)
	at cljs.analyzer$error.invoke(analyzer.cljc:523)
	at cljs.analyzer$basic_validate_ns_spec.invoke(analyzer.cljc:1674)
etc...
```

Clearly this `load-buffer` request should only be dispatched to the clojure repl and not the cljs repl.

**Comment by Malabarba** [comment]
What about regular eval commands? Like `C-x C-e`.
Do they work?

**Comment by Malabarba** [comment]
@expez Try replacing `(nrepl-current-session)` with `(cider-current-session)` inside `cider-request:load-file`.

If that works, we might want to do that for all instances of `nrepl-current-session`.

**Comment by expez** [comment]
> What about regular eval commands? Like C-x C-e.
> Do they work?

<kbd>C-c C-c</kbd> and <kbd>C-x C-e</kbd>  works.

using `(cider-current-session)` prevents the stacktrace in the cljs repl.

**Comment by Malabarba** [comment]
Then that's probably it.
To be honest though, I have no idea how a single message would end up being sent to two repls. If that's really the case, then there could be something more subtle at play here.

**Comment by bbatsov** [comment]
Btw, there's a bigger issue at hand here - `nrepl-current-session` is used in a lot of places, where it shouldn't be. I'll go over the code and address those problems later today.

**Comment by expez** [comment]
:+1:

**Comment by Malabarba** [comment]
:+1:
(hopefully I'll be able to write more code again soon.)

**Comment by Malabarba** [comment]
>  nrepl-current-session is used in a lot of places, where it shouldn't be

When I implemented that sibling connections thing I went through all occurrences of this function. I replaced it with `cider-current-session` wherever I was sure that was the right thing. When in doubt, I just left it as is. I just had no idea how would non-eval operations (like completion or eldoc) work with the cljs repl.
Are they handled fine? Or do they have to be directed at the clj repl?
